app/models/order_item.rb

```diff
@@ -31,4 +31,11 @@ def update_message(level, message)
     progress_messages << ProgressMessage.new(:level => level, :message => message, :tenant_id => self.tenant_id)
     touch
   end
+
+  def service_parameters=(values)
```

Comment:
I can't find where this method is used, is this just done on the create in the background? Do we anticipate that we will use it by direct assignment like this in the future or would it be safer to move this to something like a `before_update`?

Reply:
I'm overriding the default rails column setter and storing the raw data in another column and the sanitized data in the original column.  This is already called when someone adds an order_item to an order.

Reply:
Right, we're not explicitly calling `service_parameters =` anywhere. That's why I'm curious if you think it might be better suited to a `before_update` or `before_save` or something like that.

I just see this from the spec side of testing this method, even though it's not really called explicitly anywhere in our code, so it looks weird. If it were in a callback, you could use the same call that we're actually doing when we add an order_item to an order and test it like that. A callback might not be the way to go, just thought the "magic" of overriding the setter was a little odd.

Reply:
Sure, I can look into moving this to a callback.  Thanks for the feedback.